index.d.ts

```diff
@@ -1,13 +1,16 @@
 declare namespace ensureError {
 	interface NonError extends Error {
 		name: 'NonError';
+		stack: string;
 	}
 }
 
 // IfAny<T, ThenType, ElseType> resolves to ThenType if T is `any` and resolves to ElseType otherwise
 // https://stackoverflow.com/a/49928360/4135063
 type IfAny<T, ThenType, ElseType> = 0 extends (1 & T) ? ThenType : ElseType;
 
+type WithStack<T> = T & { stack: string; };
```

Comment:
```suggestion
type ErrorWithStack<T> = T & {stack: string};
```

Reply:
I don't think this needs to be in the namespace. It can just be top-level in the file.

Reply:
It's not in the namespace, is it?

Reply:
Never mind. It looked like it was in the diff on the iOS GitHub app.